symcache/src/writer.rs

```diff
@@ -25,29 +25,14 @@ use utils::shorten_filename;
 ///
 /// This requires the writer to be seekable.
 pub fn to_writer<W: Write + Seek>(mut w: W, obj: &Object) -> Result<()> {
-    w.write_all(CacheFileHeader::default().as_bytes())?;
-    let header = {
-        let mut writer = SymCacheWriter::new(&mut w);
-        writer.write_object(obj)?;
-        writer.header
-    };
-    w.seek(SeekFrom::Start(0))?;
-    w.write_all(header.as_bytes())?;
-    Ok(())
+    SymCacheWriter::new(&mut w).write_object(obj)
 }
 
 /// Converts an object into a vector of symcache data.
 pub fn to_vec(obj: &Object) -> Result<Vec<u8>> {
-    let mut buf = Vec::<u8>::new();
-    buf.write_all(CacheFileHeader::default().as_bytes())?;
-    let header = {
-        let mut writer = SymCacheWriter::new(&mut buf);
-        writer.write_object(obj)?;
-        writer.header
-    };
-    let header_bytes = header.as_bytes();
-    (&mut buf[..header_bytes.len()]).copy_from_slice(header_bytes);
-    Ok(buf)
+    let mut cursor = Cursor::new(Vec::new());
```

Comment:
AFAIK we need to revert this to original code because this has a massive performance degradation.

Reply:
Just ran a couple of manual benchmarks, no measurable difference in performance. For some reason, the `Cursor`-based implementation was even 3-4% faster, but I think that is just an error of measurement.